client/me/security-2fa-setup/index.jsx

```diff
@@ -81,7 +85,10 @@ class Security2faSetup extends Component {
 				) : null }
 
 				{ 'backup-codes' === this.state.step ? (
-					<Security2faSetupBackupCodes onFinished={ this.onFinished } />
+					<Security2faSetupBackupCodes
+						doSMSFlow={ this.state.authMethod !== 'app-based' }
```

Comment:
Also, in case we add a third means of 2FA, should we be more precise here and instead of checking if we're not using app-based, check if we're on the SMS flow?

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/57434/commits/29c2e206070a1c7bd60b28cbed57278c8db9df23